### ISSUE TITLE
Movement Feedback

### ISSUE THREAD

**Issue by Wulfre** [Ketei/UnnamedGame#15]
- Increase minimum jump velocity? Jump feels inconsistent and hard to predict when jumping multiple times in a row, for example going across consecutive horizontal platforms. Bound too closely to the length of the key press, expect being able to tap the key and get decent movement.

- Decrease gravity? Character feels very heavy, and seems to fall slightly faster than jump, which makes the jump arc asymmetrical and not intuitive to judge how long to hold jump for.

- Jumping under a platform rapidly multiple times seems to build up downwards velocity and can give a large speed boost if your final jump is close enough to the corner of the platform. Can't tell if it's due to coyote or not.

- I expect walking jump to have a lower max height than running jump. I'm also used to holding down a key to change speed, rather than toggle, but that's personal preference.

- I saw someone else say that it feels like wall jumping should be possible, and I agree that I expected to be able to wall jump, at least with the layout of the test map with, especially this area:
![image](https://github.com/Ketei/UnnamedGame/assets/6633817/3d18ecb5-2168-42a6-a24e-1183706a875f)

**Comment by Ketei** [comment]
- **Hold down a key to walk:** I've added the feature with the pull request #19 Use the enter key to toggle between hold-to-walk and press-to-walk

- **Increase minimum jump velocity:** I've increased the minimum jump velocity to be around 50% of the full jump.

- **Decrease gravity:** Gravity is related to time (_time_to_floor). I think I increased it so it SHOULD feel a bit less heavy.

- **Walk jump < Run jump:** I'll make a branch where this is possible. I don't want to commit to that but I'll make it possible. Check the variable-jump-height branch. It's up to date with main but has this feature included

- **Wall jump:"" There is going to be something like wall-jump ((l)edge grab). The layout is mostly just to force you to go to the rigth in the test map

- **About downwards velocity:** This is due to how the engine handles physics. Long story short, when you hit your head your upward momentum is immediatly set to 0 and then gravity is added. If you hit a corner the physics engine then calculates your new momentum and boosts you to a side. This is because the player collision is kind of a pill. and ground is a square. (Image to show)

![Feets](https://github.com/Ketei/UnnamedGame/assets/137458469/a3d5e6bf-2bcb-4262-99cb-d13bb46de78e)
